src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequence.tsx

```diff
@@ -132,6 +141,17 @@ const BlastInputSequence = (props: Props) => {
     ? checkSequenceValidity(input, sequenceType)
     : true; // treat empty input as valid
 
+  const { sequences } = useBlastForm();
```

Comment:
I'm confused. Why does a single sequence box need access to the full list of sequences? Why doesn't it just validate its own input?

Reply:
well it only needs to know how many sequences we have to deal with the empty input sequence. remember we will know that there is an empty input sequence which is valid by looking at the sequences.length

Reply:
Ok i manage to move it further up in the code. Please have a look and make sure i didnt break any flow (specially the function to check the sequence validity).

Reply:
This is not relevant anymore since the code has been moved out of BlastInputSequence